node/src/blockchain/blockchain_interface/mod.rs

```diff
@@ -35,5 +41,19 @@ pub trait BlockchainInterface {
         consuming_wallet: Wallet,
     ) -> Box<dyn Future<Item = Box<dyn BlockchainAgent>, Error = BlockchainAgentBuildError>>;
 
+    fn process_transaction_receipts(
+        &self,
+        transaction_hashes: Vec<H256>,
+    ) -> Box<dyn Future<Item = Vec<TransactionReceiptResult>, Error = BlockchainError>>;
+
+    fn submit_payables_in_batch(
+        &self,
+        logger: Logger,
+        chain: Chain,
+        agent: Box<dyn BlockchainAgent>,
```

Comment:
It would be quite cool, if the agent could simply answer what the chain is. This agent should know about the details around the blockchain as much as nobody else, therefore it looks to me a bit ugly if I see that there are two arguments, the agent and the chain, instead of passing in only one, the agent.

If you think you don't have time to do it, I'll understand.